### PR TITLE
Add AutoBot type and queries to GraphQL 

### LINKED ISSUE #196: AutoBot oAuth query and mutation
autobot request to db (slack_team_id) -> matches the AutoBot Table row for that slack_team_id -> gets the cohort based on the cohort_id column of that row

### PR DESCRIPTION
**Pull request by the-vampiire**
closes #196

### CODE REVIEW

**Comment by the-vampiire** [comment]
getting this error message when trying to run some GiQL tests @TheSabby @Oxyrus 

```
/Users/vamp/Documents/Code/Chingu/chingu-api/node_modules/sequelize/lib/associations/belongs-to-many.js:63
      throw new AssociationError('belongsToMany must be given a through option, either a string or a model');
      ^

SequelizeAssociationError: belongsToMany must be given a through option, either a string or a model
    at new BelongsToMany (/Users/vamp/Documents/Code/Chingu/chingu-api/node_modules/sequelize/lib/associations/belongs-to-many.js:63:13)
    at Function.belongsToMany (/Users/vamp/Documents/Code/Chingu/chingu-api/node_modules/sequelize/lib/associations/mixin.js:50:25)
    at Function.CohortTeamTierAct.associate (/Users/vamp/Documents/Code/Chingu/chingu-api/models/cohort_team_tier_act.js:40:23)
    at Object.keys.forEach (/Users/vamp/Documents/Code/Chingu/chingu-api/models/index.js:29:19)
    at Array.forEach (<anonymous>)
    at Object.<anonymous> (/Users/vamp/Documents/Code/Chingu/chingu-api/models/index.js:27:17)
    at Module._compile (module.js:635:30)
    at Object.Module._extensions..js (module.js:646:10)
    at Module.load (module.js:554:32)
    at tryModuleLoad (module.js:497:12)
    at Function.Module._load (module.js:489:3)
    at Module.require (module.js:579:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/vamp/Documents/Code/Chingu/chingu-api/config/auth.js:2:18)
    at Module._compile (module.js:635:30)
    at Object.Module._extensions..js (module.js:646:10)
    at Module.load (module.js:554:32)
    at tryModuleLoad (module.js:497:12)
    at Function.Module._load (module.js:489:3)
    at Module.require (module.js:579:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/vamp/Documents/Code/Chingu/chingu-api/app.js:6:26)
[nodemon] app crashed - waiting for file changes before starting...
```

File: schema/resolvers.js [hunk 1]
```diff
@@ -19,6 +19,12 @@ module.exports = {
 
     group: async (root, { group_id }, { models: { Group } }) => Group.findById(group_id),
 
+    autobot: async (root, { slack_team_id }, { models: { AutoBot } }) => {
+      // handle autobot_token (grab from context)
```
Comment (TheSabby) [inline schema/resolvers.js]: without the return statement this wouldn't work. What is the linter saying?

File: schema/type-defs.js [hunk 2]
```diff
@@ -61,6 +61,14 @@ module.exports = `
     tiers: [Tier!]!
   }
 
+  type AutoBot {
+    slack_team_id: String!
+    slack_team_token: String!
+    autobot_id: String!
+    autobot_token: String!
+    cohort: Cohort!
```
Comment (TheSabby) [inline schema/type-defs.js]: In the model you called these `bot_id` and `bot_token`. Be consistent or add a custom resolver for them. Better to just be consistent

File: schema/type-defs.js [hunk 3]
```diff
@@ -204,6 +212,7 @@ module.exports = `
     group(group_id: ID!): Group
     city(city_id: ID!): City
     country(country_id: ID!): Country
+    autobot(slack_team_id: String!): AutoBot
```
Comment (TheSabby) [inline schema/type-defs.js]: The returning of an autobot is non-nullable

**Pull request merged**